protostar/upgrader/upgrade_manager.py

```diff
@@ -0,0 +1,189 @@
+# pylint: disable=broad-except
+import os
+import shutil
+import tarfile
+from logging import Logger
+from pathlib import Path
+from typing import Any
+
+import requests
+
+from protostar.upgrader.upgrade_remote_checker import UpgradeRemoteChecker
+from protostar.utils.protostar_directory import (
+    ProtostarDirectory,
+    VersionManager,
+    VersionType,
+)
+
+
+class UpgradeManagerException(Exception):
+    pass
+
+
+class UpgradeManager:
```

Comment:
This class also looks like it could just be a function `upgrade`?

Reply:
All dependencies of `UpgradeManager` would become dependencies of `UpgradeCommand` (and other classes that could possible use this function) and that generates boilerplate.